gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -788,7 +794,11 @@ public EnumTypeAdapter(Class<T> classOfT) {
           nameToConstant.put(name, constant);
           constantToName.put(constant, name);
         }
-      } catch (NoSuchFieldException e) {
+      } catch (IllegalAccessException e) {
+        throw new AssertionError(e);
+      } catch (NullPointerException e) {
```

Comment:
Why catch this?

Reply:
> `NullPointerException` - if the specified object is null and the field is an instance field.

But the enum constants are `static` fields so this should not be possible.

Reply:
Fixed at eba1415f1c20a225e528aac83913c3a09962c282